src/main/java/com/singh/astha/notification/service/producers/KafkaProducer.java

```diff
@@ -0,0 +1,30 @@
+package com.singh.astha.notification.service.producers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.singh.astha.notification.service.dtos.common.NotificationRequest;
+import com.singh.astha.notification.service.utils.Constants;
+import com.singh.astha.notification.service.utils.JsonUtils;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class KafkaProducer {
+
+    private final KafkaTemplate<String, String> kafkaTemplate;
+
+    private final JsonUtils jsonUtils;
+
+    public KafkaProducer(
+            KafkaTemplate<String, String> kafkaTemplate,
+            JsonUtils jsonUtils
+    ) {
+        this.kafkaTemplate = kafkaTemplate;
+        this.jsonUtils = jsonUtils;
+    }
+
+    public void pushEvent(NotificationRequest notificationRequest) throws JsonProcessingException {
```

Comment:
First: Creating such a specific method like this one(which takes a certain type as input and send to an hardcoded topic) undermines the purpose of creating a generic class.

Second: Even if you make this class generic to resolve the first issue, still it won't serve any special use case. Better is to use the two lines of code directly in the controller.